packages/gitbook/src/components/SiteSectionTabs/SiteSectionTabs.tsx

```diff
@@ -22,34 +22,36 @@ export function SiteSectionTabs(props: { sections: SiteSection[]; section: SiteS
     const currentTabRef = React.useRef<HTMLAnchorElement>(null);
     const navRef = React.useRef<HTMLDivElement>(null);
 
-    const [currentIndex, setCurrentIndex] = React.useState(
-        sections.findIndex((section) => section.id === currentSection?.id),
-    );
+    // we don't set the current tab with a click event because the tab will navigate to a different section and trigger a page reload.
+    const currentIndex = sections.findIndex((section) => section.id === currentSection?.id);
```

Comment:
We can compute the index beforehand where we fetch the structure like `sections: { list: [...], section: { index: number, object: SiteSection }`